subsys/net/ip/icmpv4.c

```diff
@@ -537,8 +538,19 @@ int net_icmpv4_send_echo_request(struct net_if *iface,
 		return -ENOMEM;
 	}
 
-	net_pkt_set_ip_dscp(pkt, net_ipv4_get_dscp(tos));
-	net_pkt_set_ip_ecn(pkt, net_ipv4_get_ecn(tos));
+#if !defined(CONFIG_NET_ALLOW_ANY_PRIORITY)
+	if (priority > NET_MAX_PRIORITIES) {
+		NET_ERR("Priority %d is too large, maximum is %d",
+			priority, NET_MAX_PRIORITIES);
+		return -EINVAL;
+	}
+#endif /* !CONFIG_NET_ALLOW_ANY_PRIORITY */
```

Comment:
The priority can't be equal to `NET_MAX_PRIORITIES` (same for `IPv6` and `zperf`).

Nits: Use `#ifndef` and add a newline after `#endif`.

```suggestion
#ifndef CONFIG_NET_ALLOW_ANY_PRIORITY
	if (priority >= NET_MAX_PRIORITIES) {
		NET_ERR("Priority %d is too large, maximum is %d",
			priority, NET_MAX_PRIORITIES);
		return -EINVAL;
	}
#endif /* !CONFIG_NET_ALLOW_ANY_PRIORITY */

```

Reply:
I think we should print the max like this

`NET_ERR("Priority %d is too large, maximum is %d", priority, NET_MAX_PRIORITIES - 1);`

if we check `priority >= NET_MAX_PRIORITIES`